File: backend/culinarycraft/culinarycraft/src/main/java/ic/project/bytebistro/culinarycraft/repository/entity/Recipe.java
package ic.project.bytebistro.culinarycraft.repository.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Entity
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "recipe")
public class Recipe {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "recipe_id")
    private Long id;

    @Column(name = "recipe_name")
    private String name;

    @Column(name = "description", length = 5000)
    private String description;

    @Column(name = "url_image")
    private String urlImage;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(
            name = "recipe_ingredients",
            joinColumns = @JoinColumn(name = "recipe_id"),
            inverseJoinColumns = @JoinColumn(name = "ingredient_id"))
    private List<Ingredient> ingredients;

    @ManyToMany(fetch = FetchType.EAGER, mappedBy = "favouritesRecipes")
    private List<User> likes;

}

File: frontend/culinary_craft/lib/Components/Recipe.dart
import 'dart:typed_data';

import 'Ingredient.dart';
import 'Like.dart';

class Recipe {
  final int id;
  final String name;
  final String description;
  final String imageURL;
  final List<Ingredient> ingredients;
  final List<Like> likes;
  final Uint8List imageData;

  Recipe( {
    required this.id,
    required this.name,
    required this.description,
    required this.imageURL,
    required this.ingredients,
    required this.likes,
    required this.imageData
  });
}
